packages/theme/src/types.ts

```diff
@@ -0,0 +1,41 @@
+export type ThemeValues = {
+  [key: string]: string | number | ThemeValues;
+};
+
+export type Theme = {
+  name: string;
+  colors: {
+    branding?: ThemeValues;
+    text?: ThemeValues;
+    background?: ThemeValues;
+    foreground?: ThemeValues;
+    border?: ThemeValues;
+  };
+  typography: { [key: string]: any }; //TODO: Update theme to support typography. Left as 'any' for legacy support until updated and migrated
+};
+
+export type ValidatedTheme = Theme & {
+  validated: true;
+};
+
+export type ReadTheme = Theme & {
+  colors: {
+    branding?: any;
+    text?: any;
+    background?: any;
+    foreground?: any;
+    border?: any;
```

Comment:
This can be an object, string, or number since it is allowed to have an arbitrary number of nesting objects. For something like `theme.colors.text.nested.nestedAgain.someColor` to work, it needs to be `any`. Otherwise, you'll get an error that `string` and `number` do not have properties `nested` or something.